### ISSUE TITLE
Christmas/New Year model performance not improving from year to year

### ISSUE THREAD

**Issue by ryankarlos** [facebook/prophet#1106]
Hello, 

I have around 5 years training data upto July 2019. I have been checking the historic performance of the model for days around christmas/new year period - by training the model upto November 2018 , November 2017, November 2016 and generating forecasts for Dec 2018, Dec 2017, Dec 2016. Ive noticed that MAPE is consistently much higher for these months (between 1.3-1.5%) compared to other months in my data (~0.5%) - especially the forecasts around the christmas/new year period seem to be overestimated by ~ 2% or so.
Ive included holiday feature for Christmas (with a window of +4, -4) and New Year (window +4, -4). Shouldn't the model be able to learn the fact that it has overestimated for 2016, 2017 for example and then reduce the value of its forecast for the December period to reduce the MAPE ?  

Also, by having the the holiday window overlapping for two different holidays (christmas and new year) - give more effect to the days in between christmas and new year ?

**Comment by bletham** [comment]
Adding a holiday is equivalent to adding a binary regressor for that day. The window adds separate regressors for each day in the window (Christmas+1 and Christmas+2 are separate binary regressors, etc.). For Dec 27 that is in both, it basically has two binary extra regressors that are 1 just for that date. This makes them unidentifiable. From a prediction perspective it is equivalent to just having a single binary extra regressor for that date, but might make the inference more finicky so I'd recommend just using (-3, +4) for New Years.

With several years of data it certainly should be able to learn the holiday effect. It does assume stationarity (holiday effect has same magnitude each year) and I'm wondering if perhaps that is what is violated. The fitted holiday effect will be something like the average across years of the holiday effect (since model fitting minimizes mean squared error, plus some regularizers).

When you say it is consistently overestimated, is that in the backtesting? If you look at the in-sample fits (model predictions on the historical dates), then I think you should find that in the history it does not consistently overestimate (like I said, it should fit something like that average effect in the history). If it is consistently overestimating in the future, then one source of this could be that the effect size is decreasing over time. Thus fitting from the history would always overestimate.

Have you tried multiplicative seasonalities? (https://facebook.github.io/prophet/docs/multiplicative_seasonality.html) . If the holiday effect is decreasing over time because it is a multiplicative effect on a decreasing baseline, then that could handle it better.

**Comment by ryankarlos** [comment]
@bletham  Thanks for the suggestions.
Yes it is being consistently overestimated in the future whilst backtesting - ie when shifting cut off dates to the previous years e.g. Nov 2018, Dec 2018, Nov 2017, Dec 2017 and forecasting with 30-60 day horizons. 
Also weirdly I have found it gets better when not including any holidays during Christmas and New Year - (MAPE improves by 0.7 % or so). There seems to be a dip and high variability across years in what Im trying to forecast during this period  

Just so I am clear regarding the holidays - would setting the limits for  New Years as (-7, 0)  be the same as having a Christmas day, Boxing day (1,0) and New Years (-4,0) ? In which case, is there any advantage in breaking the Christmas holidays into seperate days rather than having a fixed seven day holiday period (25 Dec -1st Jan) ? Is there way to check the effect of these holidays separately besides the overall holiday component ?

**Comment by bletham** [comment]
Yes, setting New Years with (-7, 0) is the same as separately setting Christmas (0, 0), Boxing day (0, 0), and New Years (-5, 0).

You can look at the effect of each holiday separately by looking at the forecast dataframe (output of  `m.predict`), where there will be a column for each holiday that has the holiday effect on each date in that dataframe (this will be 0 for all of the rows that are not the holiday date). If you do New Years (-7, 0) and made predictions with daily frequency, then you will find that there will be a column named "New Years" and each year it will have 8 non-zero entries, which will be that entire week from Christmas through New Years. If you do Christmas (0, 0) then you will find a column named "Christmas" that is non-zero only on Christmas.

I think the reason to prefer breaking the holidays out vs. having just New Years (-7, 0) would just be if you wanted an easier way to pull out just the effect on e.g. Christmas day by having it as its own column in the forecast dataframe. From the point of view of model fitting / predictions, they're entirely equivalent.
